filebeat/service.sls

```diff
@@ -1,22 +1,9 @@
-# filebeat 1.0.0 will not start without tty. use_vt in cmd.run, or sudo with !requiretty in sudoers (default) does not work.
-# this is a hack to get around that issue. 
-filebeat.sshkeygen:
-  cmd.run:
-    - name: ssh-keygen -f /root/.ssh/filebeat -P ""
-    - unless: 
-      - ls /root/.ssh/filebeat
+# -*- coding: utf-8 -*-
+# vim: ft=sls
 
-filebeat.pubkeytoauth:
-  cmd.run:
-    - name: cat /root/.ssh/filebeat.pub >> /root/.ssh/authorized_keys
-    - unless: cat /root/.ssh/filebeat.pub | grep -f - /root/.ssh/authorized_keys
-    - require:
-      - cmd: filebeat.sshkeygen
+{% from "filebeat/map.jinja" import filebeat with context %}
 
-filebeat.service:
-  cmd.run:
-    - name: ssh -t -t -o NoHostAuthenticationForLocalhost=yes -i /root/.ssh/filebeat root@localhost "su -c 'service filebeat restart'"
```

Comment:
Okay cool, this: https://github.com/saltstack-formulas/filebeat-formula/pull/5#issuecomment-203574055 is where we had been discussing the issue previously.